src/contracts/fund/trading/Trading.ts

```diff
@@ -213,13 +213,14 @@ export class Trading extends Contract {
       throw new InvalidExchangeIndexError(args.exchangeIndex);
     }
 
-    const adapterMethodIsAllowed = await registry.isAdapterMethodAllowed(exchange.adapter, args.methodSignature);
+    const encodedSignature = keccak256(args.methodSignature).substr(0, 10);
```

Comment:
What's the 10 character limit here used for? Is that a fixed n umber that we can rely on or is this just arbitrary and only works in thise specific case? Is there a better way to compute this?